client/dashboard/me/monetize-subscriptions/monetize-subscription.tsx

```diff
@@ -0,0 +1,364 @@
+									to: addFlashMessage( getMonetizeSubscriptionsUrl() ),
+								} );
+							},
+							onError: () => {
+								if ( isProduct ) {
+									createErrorNotice( __( 'There was a problem while removing your product.' ), {
+										actions: [
+											{
+												url: CALYPSO_CONTACT,
+												label: __( 'Please contact support' ),
+											},
+										],
+									} );
+								} else {
+									createErrorNotice(
+										__( 'There was a problem while stopping your subscription.' ),
+										{
+											actions: [
+												{
+													url: CALYPSO_CONTACT,
+													label: __( 'Please contact support' ),
+												},
+											],
+										}
+									);
+								}
+							},
+						} );
+					} }
+				>
+					{ title }
+				</Button>
+			}
+		/>
+	);
+}
+
+const BackButton = () => {
+	const router = useRouter();
+
+	return (
+		<Button
+			className="dashboard-page-header__back-button"
+			icon={ isRTL() ? chevronRight : chevronLeft }
+			onClick={ () => {
+				router.navigate( {
+					to: getMonetizeSubscriptionsUrl(),
+				} );
+			} }
+		>
+			{ getMonetizeSubscriptionsPageTitle() }
+		</Button>
+	);
+};
+
+export default function MonetizeSubscriptionDtails() {
+	const params = monetizeSubscriptionRoute.useParams();
+	const subscriptionId: string = params.subscriptionId;
+
+	const { data: subscription } = useQuery( monetizeSubscriptionQuery( subscriptionId ) );
+
+	const { mutate: stopSubscription, isPending: isStoppingSubscription } = useMutation(
+		monetizeSubscriptionStop( subscriptionId )
+	);
+
+	const { mutate: disableAutoRenew, isPending: isDisablingAutoRenew } = useMutation(
+		monetizeSubscriptionDisableAutoRenew( subscriptionId )
+	);
+
+	const { mutate: enableAutoRenew, isPending: isEnablingAutoRenew } = useMutation(
+		monetizeSubscriptionResumeAutoRenew( subscriptionId )
+	);
+
+	const isRenewable = subscription && ( subscription.renew_interval || subscription.is_renewable ); // can remove renew_interval once backend is deployed
+	const isAutoRenewing = isRenewable && subscription.renew_interval;
+	const isProduct = !! subscription && ! isRenewable;
+	const isUpdating = isEnablingAutoRenew || isDisablingAutoRenew;
+	const formattedExpiry = useFormattedTime( subscription?.end_date ?? '' );
+	const formattedRenewal = useFormattedTime( subscription?.end_date ?? '' );
+	const isOneTimePurchase = subscription?.renew_interval === 'one-time';
+	const expiryDateTitle = ( () => {
+		if ( isProduct ) {
+			return __( 'Paid until' );
+		}
+		if ( isAutoRenewing ) {
+			return __( 'Renews' );
+		}
+		return __( 'Expires' );
+	} )();
+
+	return (
+		<PageLayout
+			size="small"
+			header={
+				<PageHeader
+					prefix={ <BackButton /> }
+					title={ isProduct ? __( 'Product Details' ) : __( 'Subscription Details' ) }
+				/>
+			}
+		>
+			{ isStoppingSubscription && (
+				<Notice status="info">
+					{ isProduct ? __( 'Removing this product' ) : __( 'Stopping this subscription' ) }
```

Comment:
These notice messages should end in periods.

```suggestion
					{ isProduct ? __( 'Removing this product.' ) : __( 'Stopping this subscription.' ) }
```

Reply:
Done in f48060f280e89f4b4d29a63173ce9a050b51fb88

Reply:
I think this updated `client/me/memberships/subscription.jsx` but not `client/dashboard/me/monetize-subscriptions/monetize-subscription.tsx`